app/tests/spec/lib/runtime-require.js

```diff
@@ -0,0 +1,55 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'chai',
+  'lib/promise',
+  'lib/runtime-require',
+  'sinon'
+], function (chai, p, runtimeRequire, sinon) {
+  'use strict';
+
+  var assert = chai.assert;
+
+  describe('lib/runtime-require', function () {
+    var sandbox;
+    var MockModule1 = {};
+    var MockModule2 = {};
+
+    beforeEach(function () {
+      sandbox = sinon.sandbox.create();
+
+      sandbox.stub(window, 'require', function (moduleList, callback) {
+        setTimeout(function () {
+          var Module = moduleList[0] === 'module1' ? MockModule1 : MockModule2;
+          callback(Module);
+        }, 10);
```

Comment:
nit: Magic Number 10, can we either have a comment, or declare this as a constant?